### PR TITLE
net: lwm2m_client_utils: Use LWM2M_*_INIT macros

### PR DESCRIPTION
**Pull request by SeppoTakalo**
Replace many of lwm2m_<thing>_init() functions with LWM2M_OBJ_INIT() or LWM2M_APP_INIT().

Now we have much less #ifdef calls in our application main.c

### CODE REVIEW

File: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst [hunk 1]
```diff
@@ -375,6 +375,10 @@ Libraries for networking
 
    * Added the :c:func:`nrf_provisioning_set_interval` function to set the interval between provisioning attempts.
 
+* :ref:`lib_lwm2m_client_utils` library:
+
+  * Following initialization functions have been deprecated :c:func:`lwm2m_init_location`, :c:func:`lwm2m_init_firmware()`, :c:func:`lwm2m_init_device()`, :c:func:`lwm2m_init_cellular_connectivity_object()`, :c:func:`lwm2m_init_connmon()` and :c:func:`lwm2m_init_image()`.
```
Comment (peknis) [inline doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst:380]:
```suggestion
  * The following initialization functions have been deprecated:

    * :c:func:`lwm2m_init_location`
    * :c:func:`lwm2m_init_firmware()`
    * :c:func:`lwm2m_init_device()`
    * :c:func:`lwm2m_init_cellular_connectivity_object()`
    * :c:func:`lwm2m_init_connmon()`
    * :c:func:`lwm2m_init_image()`
```
For better readability.

File: samples/cellular/lwm2m_client/fota.rst [hunk 2]
```diff
@@ -97,9 +97,8 @@ Initializing
 
 The Advanced Firmware Update functionality is implemented in :file:`lwm2m_firmware.c` and :file:`lwm2m_adv_firmware.c`.
 The Advanced Firmware Update object shares the same API with the standard LwM2M Firmware Update object as implemented in the :ref:`lib_lwm2m_client_utils` library.
-This object is initialized by calling the :c:func:`lwm2m_init_firmware` function and by confirming the currently booted application image as valid by calling the :c:func:`lwm2m_init_image` function.
-The validation of the image marks the FOTA process as complete, and the result is reported to the server.
-The modem FOTA process is automatically validated.
+The image automatically is confirmed to be valid on the boot and that marks the FOTA process as completed.
+The modem FOTA process is automatically validated during update, it does not reboot the device.
```
Comment (peknis) [inline samples/cellular/lwm2m_client/fota.rst:101]:
```suggestion
The image is automatically confirmed to be valid on the boot and that marks the FOTA process as completed.
The modem FOTA process is automatically validated during update; it does not reboot the device.
```

**Review by peknis — CHANGES_REQUESTED** [review]

**Review by peknis — APPROVED** [review]

File: include/net/lwm2m_client_utils.h [hunk 3]
```diff
@@ -411,6 +374,20 @@ int lwm2m_adv_firmware_create_inst(const char *component,
 #define LWM2M_ADV_FOTA_LINKED_INSTANCES_ID 16
 #define LWM2M_ADV_FOTA_CONFLICTING_INSTANCES_ID 17
 
+/********** DEPRECATED FUNCTIONS *************/
+/** @deprecated */
+static inline int lwm2m_init_firmware(void) {return 0; }
```
[the remark this answers is not in the document]
Comment (SeppoTakalo) [reply include/net/lwm2m_client_utils.h:379]: What would be the point of breaking the existing application by asserting or failing the build?

I think that would be API break and should not be allowed on public API.
This deprecation and replacing with  dummy stubs leaves user more time to refactor their application.

[the remark this answers is not in the document]
Comment (MarkusLassila) [reply include/net/lwm2m_client_utils.h:379]: Ah, now I see. The inits are no longer necessary at all as the initialization is done with the with the init_ macros. I got stuck on the idea that we should still call the underlying implementation from these.

**Review by MarkusLassila — APPROVED** [review]

**Review by juhaylinen — APPROVED** [review]

**Review by rlubos — APPROVED** [review]

**Review by nordicjm — APPROVED** [review]
Sample changes OK

**Comment by rlubos** [comment]
@SeppoTakalo Changelog conflict :(

File: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst [hunk 4]
```diff
@@ -388,6 +388,17 @@ Libraries for networking
 
   * Updated to request proprietary PSM mode for ``SOC_NRF9151_LACA`` and ``SOC_NRF9131_LACA`` in addition to ``SOC_NRF9161_LACA``.
 
+* :ref:`lib_lwm2m_client_utils` library:
+
+  * The following initialization functions have been deprecated:
```
Comment (trantanen) [inline doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst:393]: We should have a comment somewhere saying what to do instead. You could mention it here. And I'm not sure if you also need a statement in the migration guide.

Comment (SeppoTakalo) [reply doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst]: Good point. I modified the text to `following initialization functions have been deprecated as these modules are now initialized automatically on boot`

And 
> :c:func:`lwm2m_init_firmware` is deprecated in favour of :c:func:`lwm2m_init_firmware_cb`

So basically there is nothing to do other than remove those obsolete _init() functions.

**Review by trantanen — APPROVED** [review]

**Pull request merged**